src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/Zaimanhua.kt

```diff
@@ -0,0 +1,225 @@
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import okhttp3.FormBody
+import okhttp3.Headers
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.security.MessageDigest
+
+class Zaimanhua : HttpSource(), ConfigurableSource {
+    override val lang = "zh"
+    override val supportsLatest = true
+
+    override val name = "再漫画"
+    override val baseUrl = "https://manhua.zaimanhua.com"
+    private val apiUrl = "https://v4api.zaimanhua.com/app/v1"
+    private val accountApiUrl = "https://account-api.zaimanhua.com/v1"
+
+    private val preferences: SharedPreferences =
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+
+    override val client: OkHttpClient =
+        network.client.newBuilder().rateLimit(5).addInterceptor(::authIntercept).build()
+
+    private fun authIntercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        if (request.url.host != "v4api.zaimanhua.com" || !request.headers["authorization"].isNullOrBlank()) {
+            return chain.proceed(request)
+        }
+
+        var token: String = preferences.getString("TOKEN", "")!!
+        if (token.isBlank() || !isValid(token)) {
+            val username = preferences.getString("USERNAME", "")!!
+            val password = preferences.getString("PASSWORD", "")!!
+            token = getToken(username, password)
+            if (token.isBlank()) {
+                preferences.edit().putString("TOKEN", "").apply()
+                preferences.edit().putString("USERNAME", "").apply()
+                preferences.edit().putString("PASSWORD", "").apply()
+                return chain.proceed(request)
+            } else {
+                preferences.edit().putString("TOKEN", token).apply()
+                apiHeaders = apiHeaders.newBuilder().setToken(token).build()
+            }
+        }
+        val authRequest = request.newBuilder().apply {
+            header("authorization", "Bearer $token")
+        }.build()
+        return chain.proceed(authRequest)
+    }
+
+    private fun Headers.Builder.setToken(token: String): Headers.Builder = apply {
+        if (token.isNotBlank()) set("authorization", "Bearer $token")
+    }
+
+    private var apiHeaders = headersBuilder().setToken(preferences.getString("TOKEN", "")!!).build()
```

Comment:
shouldn't set blank authorization token

Reply:
I have checked `token.isNotBlank` in  function `setToken`

Reply:
oh right...